samples/nrf_desktop/src/modules/config.c

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <stdlib.h>
+
+#include <zephyr.h>
+#include <settings/settings.h>
+
+#include "event_manager.h"
+#include "config_event.h"
+
+#define MODULE config
+#include "module_state_event.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_CONFIG_LOG_LEVEL);
+
+#define KEY_LEN 9
+
+/* Array of pointers to data loaded from settings */
+static struct config_event *loaded_data[CONFIG_EVENT_ID_MAX];
+
+static bool config_id_is_supported(u8_t id)
+{
+	if (IS_ENABLED(CONFIG_DESKTOP_MOTION_OPTICAL_ENABLE)) {
+		if ((id >= CONFIG_EVENT_ID_MOTION_CPI) &&
+		    (id <= CONFIG_EVENT_ID_MOTION_DOWNSHIFT_REST2)) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+static int settings_set(int argc, char **argv, void *val_ctx)
+{
+	if (argc != 1) {
+		return -ENOENT;
+	}
+
+	char *end;
+	u8_t id = strtol(argv[0], &end, 10);
```

Comment:
case when value returned by `strtol` is greater then 255 is not checked
note that this can be dangerous as 256 cast to `u8_t` will be 0

Reply:
Fixed